src/collapse/vsprof.rs

```diff
@@ -0,0 +1,188 @@
+
+/// A stack collapser for the output of the built in Visual Studio profiler.
+#[derive(Default)]
+pub struct Folder {
+    /// Function entries on the stack in this entry thus far.
+    stack: Vec<(String, usize)>,
+}
+
+impl Collapse for Folder {
+    fn collapse<R, W>(&mut self, mut reader: R, writer: W) -> io::Result<()>
+    where
+        R: std::io::BufRead,
+        W: std::io::Write,
+    {
+        // Skip the header
+        let mut line = Vec::new();
+        if reader.read_until(b'\n', &mut line)? == 0 {
+            warn!("File ended before start of call graph");
+            return Ok(());
+        };
+
+        // Process the data
+        let mut occurences = Occurrences::new(1);
+        loop {
+            line.clear();
+            if reader.read_until(b'\n', &mut line)? == 0 {
+                break;
+            }
+            let l = String::from_utf8_lossy(&line);
+            let line = l.trim_end();
+            if line.is_empty() {
+                continue;
+            } else {
+                self.on_line(line, &mut occurences)?;
+            }
+        }
+
+        // Write the results
+        occurences.write_and_clear(writer)?;
+
+        // Reset the state
+        self.stack.clear();
+        Ok(())
+    }
+
+    fn is_applicable(&mut self, input: &str) -> Option<bool> {
+        let mut input = input.as_bytes();
+        let mut line = String::new();
+        if let Ok(n) = input.read_line(&mut line) {
+            if n == 0 {
+                return Some(false);
+            }
+        } else {
+            return Some(false);
+        }
+
+        // Some files may start with the <U+FEFF> character (zero width no-break space). This
+        // causes the call to `starts_with` to return false, which in this case isn't what we want.
+        // As this character has no influence on the rest of the file, we can safely ignore it.
+        if line.starts_with(START_LINE) || line.starts_with(&("\u{feff}".to_owned() + START_LINE)) {
+            Some(true)
+        } else {
+            Some(false)
+        }
+    }
+}
+
+impl Folder {
+    // Parse lines like:
+    // Level,Function Name,Number of Calls,...
+    // 6,"System.String.IsNullOrEmpty(string)",4,0.00,0.00,0.00,0.00,"mscorlib.dll",
+    fn on_line(&mut self, line: &str, occurences: &mut Occurrences) -> io::Result<()> {
+        let (depth, remainder) = get_next_number(line)?;
+        // After the depth is parsed from `line`, it still has a leading comma and double
+        // quote. By trimming those, we know that the first double quote is the closing double
+        // quote surrounding the function name. By splitting on this double quote, we get the
+        // function name and the remainder of the line.
+        let x: &[_] = &[',', '"'];
+        let mut split = remainder.trim_start_matches(x).splitn(2, '"');
```

Comment:
Here, too, `split_once`. It also simplifies your `match` below slightly, and you can just make it
```rust
if let Some((function_name, remainder)) = split
```